File: tui/src/scala/tui/widgets/canvas/Rectangle.scala
package tui
package widgets
package canvas

/** Shape to draw a rectangle from a `Rect` with the given color
  */
case class Rectangle(
    x: Double,
    y: Double,
    width: Double,
    height: Double,
    color: Color
) extends Shape {
  override def draw(painter: Painter): Unit = {
    val lines = Array(
      Line(x1 = x, y1 = y, x2 = x, y2 = y + height, color = color),
      Line(x1 = x, y1 = y + height, x2 = x + width, y2 = y + height, color = color),
      Line(x1 = x + width, y1 = y, x2 = x + width, y2 = y + height, color = color),
      Line(x1 = x, y1 = y, x2 = x + width, y2 = y, color = color)
    )
    lines.foreach { line =>
      line.draw(painter)
    }
  }
}

File: tui/src/java/tui/widgets/canvas/Rectangle.java
package tui.widgets.canvas;

import tui.Color;

/// Shape to draw a rectangle from a `Rect` with the given color
public final class Rectangle implements Shape {
  public final double x;
  public final double y;
  public final double width;
  public final double height;
  public final Color color;

  public Rectangle(double x, double y, double width, double height, Color color) {
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
    this.color = color;
  }

  @Override
  public void draw(Painter painter) {
    Line[] lines =
        new Line[] {
          new Line(x, y, x, y + height, color),
          new Line(x, y + height, x + width, y + height, color),
          new Line(x + width, y, x + width, y + height, color),
          new Line(x, y, x + width, y, color)
        };
    for (Line line : lines) {
      line.draw(painter);
    }
  }
}
